### PR TITLE
SI-1099 improve docusign errors

### PR DESCRIPTION
**Pull request by danchapman**

### CODE REVIEW

File: lib/hancock/request.rb [hunk 1]
```diff
@@ -123,18 +132,19 @@ def response_body
...
-        false
+    def error_code
+      nested_hash_value(parsed_response, "errorCode")
+    end
+
+    def message
+      nested_hash_value(parsed_response, "message")
+    end
+
+    def nested_hash_value(obj, key)
```
Comment (natemontgomery) [inline lib/hancock/request.rb]: I'm not sure we want to create a new method for this, lets discuss.

File: lib/hancock/request.rb [hunk 2]
```diff
@@ -3,7 +3,16 @@
 
 module Hancock
   class Request
-    RequestError = Class.new(StandardError)
+    class RequestError < StandardError
```
Comment (natemontgomery) [inline lib/hancock/request.rb]: I would rather have this error class defined in its own file.  Especially now that it has unique behavior.

Comment (danchapman) [inline lib/hancock/request.rb]: I was wondering about that, sure let's talk about where that should be

File: lib/hancock/request.rb (hunk 1 above)
Comment (danchapman) [inline lib/hancock/request.rb]: I felt the same way, and could not find a great way to resolve it. Let's talk for sure.

File: lib/hancock/request.rb [hunk 3]
```diff
@@ -123,19 +121,29 @@ def response_body
...
-    def includes_error_code?(data)
+    def error_code
+      find_by_key(parsed_response, "errorCode")
+    end
+
+    def message
+      find_by_key(parsed_response, "message")
+    end
+
+    def find_by_key(data, key)
```
Comment (natemontgomery) [inline lib/hancock/request.rb]: I would like to have more descriptive variable names for 'data' and 'key', what do these represent?

File: lib/hancock/request.rb [hunk 4]
```diff
@@ -123,19 +121,29 @@ def response_body
...
       case data
       when Hash
-        if data.has_key?("errorCode") && data["errorCode"] != "SUCCESS"
-          true
+        if data.has_key?(key)
```
Comment (natemontgomery) [inline lib/hancock/request.rb]: `key?` is considered more idiomatic by many (but still a Ruby community debate).  But I think we can even get more idiomatic with something like `data[key] || iteratively_find_by_key(data.values, key)`.

File: lib/hancock/request.rb (hunk 3 above)
Comment (natemontgomery) [inline lib/hancock/request.rb]: Also, I think the method `deep_find_by_key` might give clearer semantics.

**Pull request merged**